17/03/16 12:43:02 WARN SparkContext: Support for Java 7 is deprecated as of Spark 2.0.0
17/03/16 12:43:02 DEBUG MutableMetricsFactory: field org.apache.hadoop.metrics2.lib.MutableRate org.apache.hadoop.security.UserGroupInformation$UgiMetrics.loginSuccess with annotation @org.apache.hadoop.metrics2.annotation.Metric(value=[Rate of successful kerberos logins and latency (milliseconds)], valueName=Time, about=, type=DEFAULT, always=false, sampleName=Ops)
17/03/16 12:43:02 DEBUG MutableMetricsFactory: field org.apache.hadoop.metrics2.lib.MutableRate org.apache.hadoop.security.UserGroupInformation$UgiMetrics.loginFailure with annotation @org.apache.hadoop.metrics2.annotation.Metric(value=[Rate of failed kerberos logins and latency (milliseconds)], valueName=Time, about=, type=DEFAULT, always=false, sampleName=Ops)
17/03/16 12:43:02 DEBUG MutableMetricsFactory: field org.apache.hadoop.metrics2.lib.MutableRate org.apache.hadoop.security.UserGroupInformation$UgiMetrics.getGroups with annotation @org.apache.hadoop.metrics2.annotation.Metric(value=[GetGroups], valueName=Time, about=, type=DEFAULT, always=false, sampleName=Ops)
17/03/16 12:43:02 DEBUG MetricsSystemImpl: UgiMetrics, User and group related metrics
17/03/16 12:43:02 DEBUG Shell: setsid exited with exit code 0
17/03/16 12:43:02 DEBUG Groups:  Creating new Groups object
17/03/16 12:43:02 DEBUG NativeCodeLoader: Trying to load the custom-built native-hadoop library...
17/03/16 12:43:02 DEBUG NativeCodeLoader: Failed to load native-hadoop with error: java.lang.UnsatisfiedLinkError: no hadoop in java.library.path
17/03/16 12:43:02 DEBUG NativeCodeLoader: java.library.path=/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
17/03/16 12:43:02 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
17/03/16 12:43:02 DEBUG PerformanceAdvisory: Falling back to shell based
17/03/16 12:43:02 DEBUG JniBasedUnixGroupsMappingWithFallback: Group mapping impl=org.apache.hadoop.security.ShellBasedUnixGroupsMapping
17/03/16 12:43:02 DEBUG Groups: Group mapping impl=org.apache.hadoop.security.JniBasedUnixGroupsMappingWithFallback; cacheTimeout=300000; warningDeltaMs=5000
17/03/16 12:43:02 DEBUG UserGroupInformation: hadoop login
17/03/16 12:43:02 DEBUG UserGroupInformation: hadoop login commit
17/03/16 12:43:02 DEBUG UserGroupInformation: using local user:UnixPrincipal: hadoop
17/03/16 12:43:02 DEBUG UserGroupInformation: Using user: "UnixPrincipal: hadoop" with name hadoop
17/03/16 12:43:02 DEBUG UserGroupInformation: User entry: "hadoop"
17/03/16 12:43:02 DEBUG UserGroupInformation: UGI loginUser:hadoop (auth:SIMPLE)
17/03/16 12:43:02 WARN SparkConf: Detected deprecated memory fraction settings: [spark.storage.memoryFraction]. As of Spark 1.6, execution and storage memory management are unified. All memory fractions used in the old model are now deprecated and no longer read. If you wish to use the old memory management, you may explicitly enable `spark.memory.useLegacyMode` (not recommended).
17/03/16 12:43:02 WARN SparkConf: 
SPARK_CLASSPATH was detected (set to '/home/hadoop/bryantchang/platforms/spark2.0/lib/mysql-connector-java-5.1.40-bin.jar:').
This is deprecated in Spark 1.0+.

Please instead use:
 - ./spark-submit with --driver-class-path to augment the driver classpath
 - spark.executor.extraClassPath to augment the executor classpath
        
17/03/16 12:43:02 WARN SparkConf: Setting 'spark.executor.extraClassPath' to '/home/hadoop/bryantchang/platforms/spark2.0/lib/mysql-connector-java-5.1.40-bin.jar:' as a work-around.
17/03/16 12:43:02 WARN SparkConf: Setting 'spark.driver.extraClassPath' to '/home/hadoop/bryantchang/platforms/spark2.0/lib/mysql-connector-java-5.1.40-bin.jar:' as a work-around.
17/03/16 12:43:02 DEBUG InternalLoggerFactory: Using SLF4J as the default logging framework
17/03/16 12:43:02 DEBUG PlatformDependent0: java.nio.Buffer.address: available
17/03/16 12:43:02 DEBUG PlatformDependent0: sun.misc.Unsafe.theUnsafe: available
17/03/16 12:43:02 DEBUG PlatformDependent0: sun.misc.Unsafe.copyMemory: available
17/03/16 12:43:02 DEBUG PlatformDependent0: direct buffer constructor: available
17/03/16 12:43:02 DEBUG PlatformDependent0: java.nio.Bits.unaligned: available, true
17/03/16 12:43:02 DEBUG PlatformDependent0: java.nio.DirectByteBuffer.<init>(long, int): available
17/03/16 12:43:02 DEBUG Cleaner0: java.nio.ByteBuffer.cleaner(): available
17/03/16 12:43:02 DEBUG PlatformDependent: Java version: 7
17/03/16 12:43:02 DEBUG PlatformDependent: -Dio.netty.noUnsafe: false
17/03/16 12:43:02 DEBUG PlatformDependent: sun.misc.Unsafe: available
17/03/16 12:43:02 DEBUG PlatformDependent: -Dio.netty.noJavassist: false
17/03/16 12:43:02 DEBUG PlatformDependent: Javassist: available
17/03/16 12:43:02 DEBUG PlatformDependent: -Dio.netty.tmpdir: /tmp (java.io.tmpdir)
17/03/16 12:43:02 DEBUG PlatformDependent: -Dio.netty.bitMode: 64 (sun.arch.data.model)
17/03/16 12:43:02 DEBUG PlatformDependent: -Dio.netty.noPreferDirect: false
17/03/16 12:43:02 DEBUG PlatformDependent: io.netty.maxDirectMemory: 1342177280 bytes
17/03/16 12:43:02 DEBUG JavassistTypeParameterMatcherGenerator: Generated: io.netty.util.internal.__matchers__.org.apache.spark.network.protocol.MessageMatcher
17/03/16 12:43:02 DEBUG JavassistTypeParameterMatcherGenerator: Generated: io.netty.util.internal.__matchers__.io.netty.buffer.ByteBufMatcher
17/03/16 12:43:02 DEBUG MultithreadEventLoopGroup: -Dio.netty.eventLoopThreads: 8
17/03/16 12:43:02 DEBUG NioEventLoop: -Dio.netty.noKeySetOptimization: false
17/03/16 12:43:02 DEBUG NioEventLoop: -Dio.netty.selectorAutoRebuildThreshold: 512
17/03/16 12:43:02 DEBUG PlatformDependent: org.jctools-core.MpscChunkedArrayQueue: available
17/03/16 12:43:03 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.numHeapArenas: 8
17/03/16 12:43:03 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.numDirectArenas: 8
17/03/16 12:43:03 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.pageSize: 8192
17/03/16 12:43:03 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.maxOrder: 11
17/03/16 12:43:03 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.chunkSize: 16777216
17/03/16 12:43:03 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.tinyCacheSize: 512
17/03/16 12:43:03 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.smallCacheSize: 256
17/03/16 12:43:03 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.normalCacheSize: 64
17/03/16 12:43:03 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.maxCachedBufferCapacity: 32768
17/03/16 12:43:03 DEBUG PooledByteBufAllocator: -Dio.netty.allocator.cacheTrimInterval: 8192
17/03/16 12:43:03 DEBUG ThreadLocalRandom: -Dio.netty.initialSeedUniquifier: 0x1cd6155d6d0e8df1 (took 0 ms)
17/03/16 12:43:03 DEBUG ByteBufUtil: -Dio.netty.allocator.type: unpooled
17/03/16 12:43:03 DEBUG ByteBufUtil: -Dio.netty.threadLocalDirectBufferSize: 65536
17/03/16 12:43:03 DEBUG ByteBufUtil: -Dio.netty.maxThreadLocalCharBufferSize: 16384
17/03/16 12:43:03 DEBUG NetUtil: Loopback interface: lo (lo, 0:0:0:0:0:0:0:1%1)
17/03/16 12:43:03 DEBUG NetUtil: /proc/sys/net/core/somaxconn: 128
17/03/16 12:43:03 DEBUG AbstractByteBuf: -Dio.netty.buffer.bytebuf.checkAccessible: true
17/03/16 12:43:03 DEBUG ResourceLeakDetector: -Dio.netty.leakDetection.level: simple
17/03/16 12:43:03 DEBUG ResourceLeakDetector: -Dio.netty.leakDetection.maxRecords: 4
17/03/16 12:43:03 DEBUG ResourceLeakDetectorFactory: Loaded default ResourceLeakDetector: io.netty.util.ResourceLeakDetector@65b89a68
17/03/16 12:43:03 DEBUG Recycler: -Dio.netty.recycler.maxCapacity.default: 32768
17/03/16 12:43:03 DEBUG Recycler: -Dio.netty.recycler.maxSharedCapacityFactor: 2
17/03/16 12:43:03 DEBUG Recycler: -Dio.netty.recycler.linkCapacity: 16
17/03/16 12:43:03 DEBUG Recycler: -Dio.netty.recycler.ratio: 8
17/03/16 12:43:03 DEBUG BlockReaderLocal: dfs.client.use.legacy.blockreader.local = false
17/03/16 12:43:03 DEBUG BlockReaderLocal: dfs.client.read.shortcircuit = false
17/03/16 12:43:03 DEBUG BlockReaderLocal: dfs.client.domain.socket.data.traffic = false
17/03/16 12:43:03 DEBUG BlockReaderLocal: dfs.domain.socket.path = 
17/03/16 12:43:03 DEBUG RetryUtils: multipleLinearRandomRetry = null
17/03/16 12:43:03 DEBUG Server: rpcKind=RPC_PROTOCOL_BUFFER, rpcRequestWrapperClass=class org.apache.hadoop.ipc.ProtobufRpcEngine$RpcRequestWrapper, rpcInvoker=org.apache.hadoop.ipc.ProtobufRpcEngine$Server$ProtoBufRpcInvoker@44edbc6d
17/03/16 12:43:03 DEBUG Client: getting client out of cache: org.apache.hadoop.ipc.Client@7739fd54
17/03/16 12:43:03 DEBUG PerformanceAdvisory: Both short-circuit local reads and UNIX domain socket are disabled.
17/03/16 12:43:03 DEBUG DataTransferSaslUtil: DataTransferProtocol not using SaslPropertiesResolver, no QOP found in configuration for dfs.data.transfer.protection
17/03/16 12:43:03 DEBUG Client: The ping interval is 60000 ms.
17/03/16 12:43:03 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/16 12:43:03 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/16 12:43:03 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #0
17/03/16 12:43:03 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #0
17/03/16 12:43:03 DEBUG ProtobufRpcEngine: Call: getFileInfo took 25ms
17/03/16 12:43:03 DEBUG DFSClient: /eventLogs/app-20170316124303-0007.lz4.inprogress: masked=rw-r--r--
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #1
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #1
17/03/16 12:43:04 DEBUG ProtobufRpcEngine: Call: create took 26ms
17/03/16 12:43:04 DEBUG DFSClient: computePacketChunkSize: src=/eventLogs/app-20170316124303-0007.lz4.inprogress, chunkSize=516, chunksPerPacket=126, packetSize=65016
17/03/16 12:43:04 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-490428901_1] with renew id 1 started
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #2
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #2
17/03/16 12:43:04 DEBUG ProtobufRpcEngine: Call: setPermission took 7ms
17/03/16 12:43:04 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=0 lastFlushOffset=0 createNewBlock=false
17/03/16 12:43:04 DEBUG DFSClient: Waiting for ack for: -1
START load
17/03/16 12:43:04 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=0 lastFlushOffset=0 createNewBlock=false
17/03/16 12:43:04 DEBUG DFSClient: Waiting for ack for: -1
START training
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #3
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #3
17/03/16 12:43:04 DEBUG ProtobufRpcEngine: Call: getFileInfo took 1ms
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #4
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #4
17/03/16 12:43:04 DEBUG ProtobufRpcEngine: Call: getListing took 1ms
17/03/16 12:43:04 DEBUG FileInputFormat: Time taken to get FileStatuses: 15
17/03/16 12:43:04 INFO FileInputFormat: Total input paths to process : 10
17/03/16 12:43:04 DEBUG FileInputFormat: Total # of splits generated by getSplits: 10, TimeTaken: 19
17/03/16 12:43:04 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=0 lastFlushOffset=0 createNewBlock=false
17/03/16 12:43:04 DEBUG DFSClient: Waiting for ack for: -1
17/03/16 12:43:04 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=0, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=0
17/03/16 12:43:04 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6270 lastFlushOffset=0 createNewBlock=false
17/03/16 12:43:04 DEBUG DFSClient: Queued packet 0
17/03/16 12:43:04 DEBUG DFSClient: Waiting for ack for: 0
17/03/16 12:43:04 DEBUG DFSClient: Allocating new block
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #5
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #5
17/03/16 12:43:04 DEBUG ProtobufRpcEngine: Call: addBlock took 4ms
17/03/16 12:43:04 DEBUG DFSClient: pipeline = DatanodeInfoWithStorage[172.21.15.173:50010,DS-bcd3842f-7acc-473a-9a24-360950418375,DISK]
17/03/16 12:43:04 DEBUG DFSClient: Connecting to datanode 172.21.15.173:50010
17/03/16 12:43:04 DEBUG DFSClient: Send buf size 124928
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #6
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #6
17/03/16 12:43:04 DEBUG ProtobufRpcEngine: Call: getServerDefaults took 1ms
17/03/16 12:43:04 DEBUG SaslDataTransferClient: SASL client skipping handshake in unsecured configuration for addr = /172.21.15.173, datanodeId = DatanodeInfoWithStorage[172.21.15.173:50010,DS-bcd3842f-7acc-473a-9a24-360950418375,DISK]
17/03/16 12:43:04 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 0 offsetInBlock: 0 lastPacketInBlock: false lastByteOffsetInBlock: 6270
17/03/16 12:43:04 DEBUG DFSClient: DFSClient seqno: 0 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #7
17/03/16 12:43:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #7
17/03/16 12:43:04 DEBUG ProtobufRpcEngine: Call: fsync took 9ms
17/03/16 12:43:04 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=1, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/16 12:43:04 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6270 lastFlushOffset=6270 createNewBlock=false
17/03/16 12:43:04 DEBUG DFSClient: Waiting for ack for: 0
17/03/16 12:43:07 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=2, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/16 12:43:07 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6270 lastFlushOffset=6270 createNewBlock=false
17/03/16 12:43:07 DEBUG DFSClient: Waiting for ack for: 0
17/03/16 12:43:07 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=3, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/16 12:43:07 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6270 lastFlushOffset=6270 createNewBlock=false
17/03/16 12:43:07 DEBUG DFSClient: Waiting for ack for: 0
17/03/16 12:43:07 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=4, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/16 12:43:07 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6270 lastFlushOffset=6270 createNewBlock=false
17/03/16 12:43:07 DEBUG DFSClient: Waiting for ack for: 0
17/03/16 12:43:07 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=5, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/16 12:43:07 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6270 lastFlushOffset=6270 createNewBlock=false
17/03/16 12:43:07 DEBUG DFSClient: Waiting for ack for: 0
17/03/16 12:43:07 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=6, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/16 12:43:07 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6270 lastFlushOffset=6270 createNewBlock=false
17/03/16 12:43:07 DEBUG DFSClient: Waiting for ack for: 0
17/03/16 12:43:07 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=7, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/16 12:43:07 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=6270 lastFlushOffset=6270 createNewBlock=false
17/03/16 12:43:07 DEBUG DFSClient: Waiting for ack for: 0
[Stage 2:=====>                                                    (1 + 0) / 10][Stage 2:=====>                                                    (1 + 4) / 10][Stage 2:=============================>                            (5 + 4) / 10]17/03/16 12:43:14 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/16 12:43:14 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
[Stage 2:==================================>                       (6 + 4) / 10][Stage 2:====================================================>     (9 + 1) / 10]17/03/16 12:43:16 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=8, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=6144
17/03/16 12:43:16 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=14007 lastFlushOffset=6270 createNewBlock=false
17/03/16 12:43:16 DEBUG DFSClient: Queued packet 8
17/03/16 12:43:16 DEBUG DFSClient: Waiting for ack for: 8
17/03/16 12:43:16 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 8 offsetInBlock: 6144 lastPacketInBlock: false lastByteOffsetInBlock: 14007
17/03/16 12:43:16 DEBUG DFSClient: DFSClient seqno: 8 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/16 12:43:17 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=9, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=13824
17/03/16 12:43:17 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=18640 lastFlushOffset=14007 createNewBlock=false
17/03/16 12:43:17 DEBUG DFSClient: Queued packet 9
17/03/16 12:43:17 DEBUG DFSClient: Waiting for ack for: 9
17/03/16 12:43:17 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 9 offsetInBlock: 13824 lastPacketInBlock: false lastByteOffsetInBlock: 18640
                                                                                17/03/16 12:43:17 DEBUG DFSClient: DFSClient seqno: 9 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/16 12:43:17 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=10, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=18432
17/03/16 12:43:17 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=18640 lastFlushOffset=18640 createNewBlock=false
17/03/16 12:43:17 DEBUG DFSClient: Waiting for ack for: 9
17/03/16 12:43:17 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=11, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=18432
17/03/16 12:43:17 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=18640 lastFlushOffset=18640 createNewBlock=false
17/03/16 12:43:17 DEBUG DFSClient: Waiting for ack for: 9
[Stage 4:>                                                         (0 + 4) / 10][Stage 4:=======================>                                  (4 + 4) / 10]17/03/16 12:43:34 DEBUG Client: The ping interval is 60000 ms.
17/03/16 12:43:34 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/16 12:43:34 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/16 12:43:34 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #8
17/03/16 12:43:34 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #8
17/03/16 12:43:34 DEBUG ProtobufRpcEngine: Call: renewLease took 6ms
17/03/16 12:43:34 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-490428901_1
17/03/16 12:43:34 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-490428901_1] with renew id 1 executed
17/03/16 12:43:44 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/16 12:43:44 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
[Stage 4:=============================>                            (5 + 4) / 10][Stage 4:==============================================>           (8 + 2) / 10]17/03/16 12:43:50 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=12, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=18432
17/03/16 12:43:50 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=23246 lastFlushOffset=18640 createNewBlock=false
17/03/16 12:43:50 DEBUG DFSClient: Queued packet 12
17/03/16 12:43:50 DEBUG DFSClient: Waiting for ack for: 12
17/03/16 12:43:50 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 12 offsetInBlock: 18432 lastPacketInBlock: false lastByteOffsetInBlock: 23246
17/03/16 12:43:50 WARN DFSClient: Slow ReadProcessor read fields took 33575ms (threshold=30000ms); ack: seqno: 12 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0, targets: [DatanodeInfoWithStorage[172.21.15.173:50010,DS-bcd3842f-7acc-473a-9a24-360950418375,DISK]]
[Stage 5:>                                                          (0 + 2) / 2]17/03/16 12:43:52 WARN TaskSetManager: Lost task 0.0 in stage 5.0 (TID 24, 172.21.15.173, executor 0): java.lang.OutOfMemoryError: Java heap space
	at java.util.Arrays.copyOf(Arrays.java:2271)
	at java.io.ByteArrayOutputStream.grow(ByteArrayOutputStream.java:118)
	at java.io.ByteArrayOutputStream.ensureCapacity(ByteArrayOutputStream.java:93)
	at java.io.ByteArrayOutputStream.write(ByteArrayOutputStream.java:153)
	at org.apache.spark.util.ByteBufferOutputStream.write(ByteBufferOutputStream.scala:41)
	at java.io.ObjectOutputStream$BlockDataOutputStream.write(ObjectOutputStream.java:1852)
	at java.io.ObjectOutputStream.write(ObjectOutputStream.java:708)
	at org.apache.spark.util.Utils$.writeByteBuffer(Utils.scala:238)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply$mcV$sp(TaskResult.scala:50)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply(TaskResult.scala:48)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply(TaskResult.scala:48)
	at org.apache.spark.util.Utils$.tryOrIOException(Utils.scala:1269)
	at org.apache.spark.scheduler.DirectTaskResult.writeExternal(TaskResult.scala:48)
	at java.io.ObjectOutputStream.writeExternalData(ObjectOutputStream.java:1458)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1429)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1177)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:347)
	at org.apache.spark.serializer.JavaSerializationStream.writeObject(JavaSerializer.scala:43)
	at org.apache.spark.serializer.JavaSerializerInstance.serialize(JavaSerializer.scala:100)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:384)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

17/03/16 12:43:52 WARN TaskSetManager: Lost task 0.1 in stage 5.0 (TID 26, 172.21.15.173, executor 0): FetchFailed(BlockManagerId(0, 172.21.15.173, 52306, None), shuffleId=1, mapId=0, reduceId=0, message=
org.apache.spark.shuffle.FetchFailedException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-4eab4354-264e-441b-823d-135e47cce24e/2b/shuffle_1_0_0.data, offset=0, length=40999142}
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.throwFetchFailedException(ShuffleBlockFetcherIterator.scala:416)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:356)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:58)
	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.util.CompletionIterator.hasNext(CompletionIterator.scala:32)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:39)
	at org.apache.spark.util.collection.ExternalAppendOnlyMap.insertAll(ExternalAppendOnlyMap.scala:154)
	at org.apache.spark.Aggregator.combineCombinersByKey(Aggregator.scala:50)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:82)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88)
	at org.apache.spark.scheduler.Task.run(Task.scala:114)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-4eab4354-264e-441b-823d-135e47cce24e/2b/shuffle_1_0_0.data, offset=0, length=40999142}
	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:113)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:349)
	... 21 more
Caused by: java.io.FileNotFoundException: /tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-4eab4354-264e-441b-823d-135e47cce24e/2b/shuffle_1_0_0.data (没有那个文件或目录)
	at java.io.FileInputStream.open(Native Method)
	at java.io.FileInputStream.<init>(FileInputStream.java:146)
	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:98)
	... 22 more

)
17/03/16 12:43:52 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=13, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=23040
17/03/16 12:43:52 WARN TaskSetManager: Lost task 1.0 in stage 5.0 (TID 25, 172.21.15.173, executor 0): TaskResultLost (result lost from block manager)
17/03/16 12:43:52 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=29771 lastFlushOffset=23246 createNewBlock=false
17/03/16 12:43:52 DEBUG DFSClient: Queued packet 13
17/03/16 12:43:52 DEBUG DFSClient: Waiting for ack for: 13
17/03/16 12:43:52 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 13 offsetInBlock: 23040 lastPacketInBlock: false lastByteOffsetInBlock: 29771
17/03/16 12:43:52 DEBUG DFSClient: DFSClient seqno: 13 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/16 12:43:52 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=14, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=29696
17/03/16 12:43:52 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=29771 lastFlushOffset=29771 createNewBlock=false
17/03/16 12:43:52 DEBUG DFSClient: Waiting for ack for: 13
17/03/16 12:43:52 WARN TransportChannelHandler: Exception in connection from /172.21.15.173:52379
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:221)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:899)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:275)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:119)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:652)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
17/03/16 12:43:52 ERROR TaskSchedulerImpl: Lost executor 0 on 172.21.15.173: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/16 12:43:52 WARN TaskSetManager: Lost task 2.0 in stage 4.1 (TID 29, 172.21.15.173, executor 0): ExecutorLostFailure (executor 0 exited caused by one of the running tasks) Reason: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/16 12:43:52 WARN TaskSetManager: Lost task 1.0 in stage 4.1 (TID 28, 172.21.15.173, executor 0): ExecutorLostFailure (executor 0 exited caused by one of the running tasks) Reason: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/16 12:43:52 WARN TaskSetManager: Lost task 0.0 in stage 4.1 (TID 27, 172.21.15.173, executor 0): ExecutorLostFailure (executor 0 exited caused by one of the running tasks) Reason: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/16 12:43:52 WARN TaskSetManager: Lost task 3.0 in stage 4.1 (TID 30, 172.21.15.173, executor 0): ExecutorLostFailure (executor 0 exited caused by one of the running tasks) Reason: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/16 12:43:52 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=15, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=29696
17/03/16 12:43:52 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=29771 lastFlushOffset=29771 createNewBlock=false
17/03/16 12:43:52 DEBUG DFSClient: Waiting for ack for: 13
[Stage 4:>                                                         (0 + 0) / 10]17/03/16 12:43:53 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=16, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=29696
17/03/16 12:43:53 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=29771 lastFlushOffset=29771 createNewBlock=false
17/03/16 12:43:53 DEBUG DFSClient: Waiting for ack for: 13
17/03/16 12:43:54 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=17, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=29696
17/03/16 12:43:54 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=29771 lastFlushOffset=29771 createNewBlock=false
17/03/16 12:43:54 DEBUG DFSClient: Waiting for ack for: 13
[Stage 4:>                                                         (0 + 4) / 10]17/03/16 12:44:04 DEBUG Client: The ping interval is 60000 ms.
17/03/16 12:44:04 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/16 12:44:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/16 12:44:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #9
17/03/16 12:44:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #9
17/03/16 12:44:04 DEBUG ProtobufRpcEngine: Call: renewLease took 8ms
17/03/16 12:44:04 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-490428901_1
17/03/16 12:44:04 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-490428901_1] with renew id 1 executed
[Stage 4:=======================>                                  (4 + 4) / 10]17/03/16 12:44:14 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/16 12:44:14 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
[Stage 4:==============================================>           (8 + 2) / 10]17/03/16 12:44:34 DEBUG Client: The ping interval is 60000 ms.
17/03/16 12:44:34 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/16 12:44:34 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/16 12:44:34 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #10
17/03/16 12:44:34 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #10
17/03/16 12:44:34 DEBUG ProtobufRpcEngine: Call: renewLease took 7ms
17/03/16 12:44:34 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-490428901_1
17/03/16 12:44:34 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-490428901_1] with renew id 1 executed
17/03/16 12:44:35 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=18, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=29696
17/03/16 12:44:35 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=39130 lastFlushOffset=29771 createNewBlock=false
17/03/16 12:44:35 DEBUG DFSClient: Queued packet 18
17/03/16 12:44:35 DEBUG DFSClient: Waiting for ack for: 18
17/03/16 12:44:35 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 18 offsetInBlock: 29696 lastPacketInBlock: false lastByteOffsetInBlock: 39130
17/03/16 12:44:35 WARN DFSClient: Slow ReadProcessor read fields took 42701ms (threshold=30000ms); ack: seqno: 18 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0, targets: [DatanodeInfoWithStorage[172.21.15.173:50010,DS-bcd3842f-7acc-473a-9a24-360950418375,DISK]]
[Stage 5:>                                                          (0 + 2) / 2]17/03/16 12:44:36 WARN TaskSetManager: Lost task 1.0 in stage 5.1 (TID 42, 172.21.15.173, executor 1): java.lang.OutOfMemoryError: Java heap space
	at java.util.Arrays.copyOf(Arrays.java:2271)
	at java.io.ByteArrayOutputStream.grow(ByteArrayOutputStream.java:118)
	at java.io.ByteArrayOutputStream.ensureCapacity(ByteArrayOutputStream.java:93)
	at java.io.ByteArrayOutputStream.write(ByteArrayOutputStream.java:153)
	at org.apache.spark.util.ByteBufferOutputStream.write(ByteBufferOutputStream.scala:41)
	at java.io.ObjectOutputStream$BlockDataOutputStream.write(ObjectOutputStream.java:1852)
	at java.io.ObjectOutputStream.write(ObjectOutputStream.java:708)
	at org.apache.spark.util.Utils$.writeByteBuffer(Utils.scala:238)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply$mcV$sp(TaskResult.scala:50)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply(TaskResult.scala:48)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply(TaskResult.scala:48)
	at org.apache.spark.util.Utils$.tryOrIOException(Utils.scala:1269)
	at org.apache.spark.scheduler.DirectTaskResult.writeExternal(TaskResult.scala:48)
	at java.io.ObjectOutputStream.writeExternalData(ObjectOutputStream.java:1458)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1429)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1177)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:347)
	at org.apache.spark.serializer.JavaSerializationStream.writeObject(JavaSerializer.scala:43)
	at org.apache.spark.serializer.JavaSerializerInstance.serialize(JavaSerializer.scala:100)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:384)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

17/03/16 12:44:36 WARN TaskSetManager: Lost task 1.1 in stage 5.1 (TID 43, 172.21.15.173, executor 1): FetchFailed(BlockManagerId(1, 172.21.15.173, 41246, None), shuffleId=1, mapId=1, reduceId=1, message=
org.apache.spark.shuffle.FetchFailedException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-6b350d52-1390-47b0-99bc-067254a42a49/0a/shuffle_1_1_0.data, offset=0, length=40999142}
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.throwFetchFailedException(ShuffleBlockFetcherIterator.scala:416)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:356)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:58)
	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.util.CompletionIterator.hasNext(CompletionIterator.scala:32)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:39)
	at org.apache.spark.util.collection.ExternalAppendOnlyMap.insertAll(ExternalAppendOnlyMap.scala:154)
	at org.apache.spark.Aggregator.combineCombinersByKey(Aggregator.scala:50)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:82)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88)
	at org.apache.spark.scheduler.Task.run(Task.scala:114)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-6b350d52-1390-47b0-99bc-067254a42a49/0a/shuffle_1_1_0.data, offset=0, length=40999142}
	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:113)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:349)
	... 21 more
Caused by: java.io.FileNotFoundException: /tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-6b350d52-1390-47b0-99bc-067254a42a49/0a/shuffle_1_1_0.data (没有那个文件或目录)
	at java.io.FileInputStream.open(Native Method)
	at java.io.FileInputStream.<init>(FileInputStream.java:146)
	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:98)
	... 22 more

)
17/03/16 12:44:36 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=19, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=38912
17/03/16 12:44:36 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=45458 lastFlushOffset=39130 createNewBlock=false
17/03/16 12:44:36 DEBUG DFSClient: Queued packet 19
17/03/16 12:44:36 DEBUG DFSClient: Waiting for ack for: 19
17/03/16 12:44:36 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 19 offsetInBlock: 38912 lastPacketInBlock: false lastByteOffsetInBlock: 45458
17/03/16 12:44:36 DEBUG DFSClient: DFSClient seqno: 19 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/16 12:44:36 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=20, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=45056
17/03/16 12:44:36 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=45458 lastFlushOffset=45458 createNewBlock=false
17/03/16 12:44:36 DEBUG DFSClient: Waiting for ack for: 19
[Stage 4:>                                                         (0 + 0) / 10]17/03/16 12:44:37 WARN TransportChannelHandler: Exception in connection from /172.21.15.173:52390
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:379)
	at io.netty.buffer.PooledUnsafeDirectByteBuf.setBytes(PooledUnsafeDirectByteBuf.java:221)
	at io.netty.buffer.AbstractByteBuf.writeBytes(AbstractByteBuf.java:899)
	at io.netty.channel.socket.nio.NioSocketChannel.doReadBytes(NioSocketChannel.java:275)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:119)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:652)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
17/03/16 12:44:37 ERROR TaskSchedulerImpl: Lost executor 1 on 172.21.15.173: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/16 12:44:37 WARN TaskSetManager: Lost task 0.0 in stage 4.2 (TID 44, 172.21.15.173, executor 1): ExecutorLostFailure (executor 1 exited caused by one of the running tasks) Reason: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/16 12:44:37 WARN TaskSetManager: Lost task 3.0 in stage 4.2 (TID 47, 172.21.15.173, executor 1): ExecutorLostFailure (executor 1 exited caused by one of the running tasks) Reason: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/16 12:44:37 WARN TaskSetManager: Lost task 2.0 in stage 4.2 (TID 46, 172.21.15.173, executor 1): ExecutorLostFailure (executor 1 exited caused by one of the running tasks) Reason: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/16 12:44:37 WARN TaskSetManager: Lost task 1.0 in stage 4.2 (TID 45, 172.21.15.173, executor 1): ExecutorLostFailure (executor 1 exited caused by one of the running tasks) Reason: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/16 12:44:37 ERROR TransportResponseHandler: Still have 1 requests outstanding when connection from /172.21.15.173:41246 is closed
17/03/16 12:44:37 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=21, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=45056
17/03/16 12:44:37 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=45458 lastFlushOffset=45458 createNewBlock=false
17/03/16 12:44:37 DEBUG DFSClient: Waiting for ack for: 19
17/03/16 12:44:38 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=22, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=45056
17/03/16 12:44:38 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=45458 lastFlushOffset=45458 createNewBlock=false
17/03/16 12:44:38 DEBUG DFSClient: Waiting for ack for: 19
17/03/16 12:44:38 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=23, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=45056
17/03/16 12:44:38 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=45458 lastFlushOffset=45458 createNewBlock=false
17/03/16 12:44:38 DEBUG DFSClient: Waiting for ack for: 19
[Stage 4:>                                                         (0 + 4) / 10]17/03/16 12:44:42 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 1 retries)
java.io.IOException: Failed to connect to /172.21.15.173:41246
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:41246
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/16 12:44:44 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/16 12:44:44 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
17/03/16 12:44:47 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 2 retries)
java.io.IOException: Failed to connect to /172.21.15.173:41246
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:41246
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/16 12:44:52 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 3 retries)
java.io.IOException: Failed to connect to /172.21.15.173:41246
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:41246
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/16 12:44:52 WARN BlockManager: Failed to fetch block after 1 fetch failures. Most recent failure cause:
org.apache.spark.SparkException: Exception thrown in awaitResult: 
	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:205)
	at org.apache.spark.network.BlockTransferService.fetchBlockSync(BlockTransferService.scala:104)
	at org.apache.spark.storage.BlockManager.getRemoteBytes(BlockManager.scala:579)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply$mcV$sp(TaskResultGetter.scala:82)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:1951)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3.run(TaskResultGetter.scala:62)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: Failed to connect to /172.21.15.173:41246
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	... 1 more
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:41246
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/16 12:44:52 WARN TaskSetManager: Lost task 0.0 in stage 5.1 (TID 41, 172.21.15.173, executor 1): TaskResultLost (result lost from block manager)
[Stage 4:=======================>                                  (4 + 4) / 10]17/03/16 12:45:04 DEBUG Client: The ping interval is 60000 ms.
17/03/16 12:45:04 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/16 12:45:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/16 12:45:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #11
17/03/16 12:45:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #11
17/03/16 12:45:04 DEBUG ProtobufRpcEngine: Call: renewLease took 7ms
17/03/16 12:45:04 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-490428901_1
17/03/16 12:45:04 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-490428901_1] with renew id 1 executed
[Stage 4:==============================================>           (8 + 2) / 10]17/03/16 12:45:14 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/16 12:45:14 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
17/03/16 12:45:19 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=24, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=45056
17/03/16 12:45:19 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=55065 lastFlushOffset=45458 createNewBlock=false
17/03/16 12:45:19 DEBUG DFSClient: Queued packet 24
17/03/16 12:45:19 DEBUG DFSClient: Waiting for ack for: 24
17/03/16 12:45:19 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 24 offsetInBlock: 45056 lastPacketInBlock: false lastByteOffsetInBlock: 55065
17/03/16 12:45:19 WARN DFSClient: Slow ReadProcessor read fields took 42813ms (threshold=30000ms); ack: seqno: 24 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0, targets: [DatanodeInfoWithStorage[172.21.15.173:50010,DS-bcd3842f-7acc-473a-9a24-360950418375,DISK]]
[Stage 5:>                                                          (0 + 2) / 2]17/03/16 12:45:21 WARN TaskSetManager: Lost task 1.0 in stage 5.2 (TID 59, 172.21.15.173, executor 2): java.lang.OutOfMemoryError: Java heap space
	at java.util.Arrays.copyOf(Arrays.java:2271)
	at java.io.ByteArrayOutputStream.grow(ByteArrayOutputStream.java:118)
	at java.io.ByteArrayOutputStream.ensureCapacity(ByteArrayOutputStream.java:93)
	at java.io.ByteArrayOutputStream.write(ByteArrayOutputStream.java:153)
	at org.apache.spark.util.ByteBufferOutputStream.write(ByteBufferOutputStream.scala:41)
	at java.io.ObjectOutputStream$BlockDataOutputStream.write(ObjectOutputStream.java:1852)
	at java.io.ObjectOutputStream.write(ObjectOutputStream.java:708)
	at org.apache.spark.util.Utils$.writeByteBuffer(Utils.scala:238)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply$mcV$sp(TaskResult.scala:50)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply(TaskResult.scala:48)
	at org.apache.spark.scheduler.DirectTaskResult$$anonfun$writeExternal$1.apply(TaskResult.scala:48)
	at org.apache.spark.util.Utils$.tryOrIOException(Utils.scala:1269)
	at org.apache.spark.scheduler.DirectTaskResult.writeExternal(TaskResult.scala:48)
	at java.io.ObjectOutputStream.writeExternalData(ObjectOutputStream.java:1458)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1429)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1177)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:347)
	at org.apache.spark.serializer.JavaSerializationStream.writeObject(JavaSerializer.scala:43)
	at org.apache.spark.serializer.JavaSerializerInstance.serialize(JavaSerializer.scala:100)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:384)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

17/03/16 12:45:21 WARN TaskSetManager: Lost task 1.1 in stage 5.2 (TID 60, 172.21.15.173, executor 2): FetchFailed(BlockManagerId(2, 172.21.15.173, 48379, None), shuffleId=1, mapId=3, reduceId=1, message=
org.apache.spark.shuffle.FetchFailedException: /tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-379b04ee-6fd6-420d-bd37-37cafa4b94f9/0c/shuffle_1_3_0.index (没有那个文件或目录)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.throwFetchFailedException(ShuffleBlockFetcherIterator.scala:416)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:392)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:58)
	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.util.CompletionIterator.hasNext(CompletionIterator.scala:32)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:39)
	at org.apache.spark.util.collection.ExternalAppendOnlyMap.insertAll(ExternalAppendOnlyMap.scala:154)
	at org.apache.spark.Aggregator.combineCombinersByKey(Aggregator.scala:50)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:82)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88)
	at org.apache.spark.scheduler.Task.run(Task.scala:114)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.FileNotFoundException: /tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-379b04ee-6fd6-420d-bd37-37cafa4b94f9/0c/shuffle_1_3_0.index (没有那个文件或目录)
	at java.io.FileInputStream.open(Native Method)
	at java.io.FileInputStream.<init>(FileInputStream.java:146)
	at org.apache.spark.shuffle.IndexShuffleBlockResolver.getBlockData(IndexShuffleBlockResolver.scala:199)
	at org.apache.spark.storage.BlockManager.getBlockData(BlockManager.scala:302)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.fetchLocalBlocks(ShuffleBlockFetcherIterator.scala:269)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.initialize(ShuffleBlockFetcherIterator.scala:303)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.<init>(ShuffleBlockFetcherIterator.scala:132)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:45)
	... 12 more

)
17/03/16 12:45:21 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=25, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=54784
17/03/16 12:45:21 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=55065 lastFlushOffset=55065 createNewBlock=false
17/03/16 12:45:21 DEBUG DFSClient: Waiting for ack for: 24
17/03/16 12:45:21 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=26, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=54784
17/03/16 12:45:21 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=55065 lastFlushOffset=55065 createNewBlock=false
17/03/16 12:45:21 DEBUG DFSClient: Waiting for ack for: 24
[Stage 4:>                                                         (0 + 0) / 10]17/03/16 12:45:22 ERROR TaskSchedulerImpl: Lost executor 2 on 172.21.15.173: Remote RPC client disassociated. Likely due to containers exceeding thresholds, or network issues. Check driver logs for WARN messages.
17/03/16 12:45:22 ERROR TransportResponseHandler: Still have 1 requests outstanding when connection from /172.21.15.173:48379 is closed
17/03/16 12:45:22 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=27, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=54784
17/03/16 12:45:22 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=61232 lastFlushOffset=55065 createNewBlock=false
17/03/16 12:45:22 DEBUG DFSClient: Queued packet 27
17/03/16 12:45:22 DEBUG DFSClient: Waiting for ack for: 27
17/03/16 12:45:22 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 27 offsetInBlock: 54784 lastPacketInBlock: false lastByteOffsetInBlock: 61232
17/03/16 12:45:22 DEBUG DFSClient: DFSClient seqno: 27 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/16 12:45:23 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=28, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=60928
17/03/16 12:45:23 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=61232 lastFlushOffset=61232 createNewBlock=false
17/03/16 12:45:23 DEBUG DFSClient: Waiting for ack for: 27
17/03/16 12:45:23 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=29, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=60928
17/03/16 12:45:23 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=61232 lastFlushOffset=61232 createNewBlock=false
17/03/16 12:45:23 DEBUG DFSClient: Waiting for ack for: 27
[Stage 4:>                                                         (0 + 4) / 10]17/03/16 12:45:27 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 1 retries)
java.io.IOException: Failed to connect to /172.21.15.173:48379
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:48379
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/16 12:45:32 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 2 retries)
java.io.IOException: Failed to connect to /172.21.15.173:48379
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:48379
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/16 12:45:34 DEBUG Client: The ping interval is 60000 ms.
17/03/16 12:45:34 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/16 12:45:34 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/16 12:45:34 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #12
17/03/16 12:45:34 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #12
17/03/16 12:45:34 DEBUG ProtobufRpcEngine: Call: renewLease took 8ms
17/03/16 12:45:34 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-490428901_1
17/03/16 12:45:34 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-490428901_1] with renew id 1 executed
17/03/16 12:45:37 ERROR RetryingBlockFetcher: Exception while beginning fetch of 1 outstanding blocks (after 3 retries)
java.io.IOException: Failed to connect to /172.21.15.173:48379
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	at java.lang.Thread.run(Thread.java:745)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:48379
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/16 12:45:37 WARN BlockManager: Failed to fetch block after 1 fetch failures. Most recent failure cause:
org.apache.spark.SparkException: Exception thrown in awaitResult: 
	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:205)
	at org.apache.spark.network.BlockTransferService.fetchBlockSync(BlockTransferService.scala:104)
	at org.apache.spark.storage.BlockManager.getRemoteBytes(BlockManager.scala:579)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply$mcV$sp(TaskResultGetter.scala:82)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3$$anonfun$run$1.apply(TaskResultGetter.scala:63)
	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:1951)
	at org.apache.spark.scheduler.TaskResultGetter$$anon$3.run(TaskResultGetter.scala:62)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: Failed to connect to /172.21.15.173:48379
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:230)
	at org.apache.spark.network.client.TransportClientFactory.createClient(TransportClientFactory.java:181)
	at org.apache.spark.network.netty.NettyBlockTransferService$$anon$1.createAndStart(NettyBlockTransferService.scala:97)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.fetchAllOutstanding(RetryingBlockFetcher.java:140)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher.access$200(RetryingBlockFetcher.java:43)
	at org.apache.spark.network.shuffle.RetryingBlockFetcher$1.run(RetryingBlockFetcher.java:170)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:144)
	... 1 more
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: 拒绝连接: /172.21.15.173:48379
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:739)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:257)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:291)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:640)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:575)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:489)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:451)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:140)
	... 2 more
17/03/16 12:45:37 WARN TaskSetManager: Lost task 0.0 in stage 5.2 (TID 58, 172.21.15.173, executor 2): TaskResultLost (result lost from block manager)
[Stage 4:=======================>                                  (4 + 4) / 10]17/03/16 12:45:44 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/16 12:45:44 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
[Stage 4:==============================================>           (8 + 2) / 10]17/03/16 12:46:04 DEBUG Client: The ping interval is 60000 ms.
17/03/16 12:46:04 DEBUG Client: Connecting to spark1/172.21.15.90:9000
17/03/16 12:46:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: starting, having connections 1
17/03/16 12:46:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #13
17/03/16 12:46:04 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #13
17/03/16 12:46:04 DEBUG ProtobufRpcEngine: Call: renewLease took 7ms
17/03/16 12:46:04 DEBUG LeaseRenewer: Lease renewed for client DFSClient_NONMAPREDUCE_-490428901_1
17/03/16 12:46:04 DEBUG LeaseRenewer: Lease renewer daemon for [DFSClient_NONMAPREDUCE_-490428901_1] with renew id 1 executed
17/03/16 12:46:04 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=30, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=60928
17/03/16 12:46:04 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=65800 lastFlushOffset=61232 createNewBlock=false
17/03/16 12:46:04 DEBUG DFSClient: Queued packet 30
17/03/16 12:46:04 DEBUG DFSClient: Waiting for ack for: 30
17/03/16 12:46:04 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 30 offsetInBlock: 60928 lastPacketInBlock: false lastByteOffsetInBlock: 65800
17/03/16 12:46:04 WARN DFSClient: Slow ReadProcessor read fields took 42353ms (threshold=30000ms); ack: seqno: 30 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0, targets: [DatanodeInfoWithStorage[172.21.15.173:50010,DS-bcd3842f-7acc-473a-9a24-360950418375,DISK]]
[Stage 5:>                                                          (0 + 2) / 2]17/03/16 12:46:05 WARN TaskSetManager: Lost task 1.0 in stage 5.3 (TID 72, 172.21.15.173, executor 3): java.lang.OutOfMemoryError: Java heap space
	at com.esotericsoftware.kryo.io.Input.readDoubles(Input.java:885)
	at com.esotericsoftware.kryo.serializers.DefaultArraySerializers$DoubleArraySerializer.read(DefaultArraySerializers.java:222)
	at com.esotericsoftware.kryo.serializers.DefaultArraySerializers$DoubleArraySerializer.read(DefaultArraySerializers.java:205)
	at com.esotericsoftware.kryo.Kryo.readObjectOrNull(Kryo.java:759)
	at com.esotericsoftware.kryo.serializers.ObjectField.read(ObjectField.java:132)
	at com.esotericsoftware.kryo.serializers.FieldSerializer.read(FieldSerializer.java:551)
	at com.esotericsoftware.kryo.Kryo.readClassAndObject(Kryo.java:790)
	at org.apache.spark.serializer.KryoDeserializationStream.readObject(KryoSerializer.scala:244)
	at org.apache.spark.serializer.DeserializationStream.readValue(Serializer.scala:159)
	at org.apache.spark.serializer.DeserializationStream$$anon$2.getNext(Serializer.scala:189)
	at org.apache.spark.serializer.DeserializationStream$$anon$2.getNext(Serializer.scala:186)
	at org.apache.spark.util.NextIterator.hasNext(NextIterator.scala:73)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:438)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.util.CompletionIterator.hasNext(CompletionIterator.scala:32)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:39)
	at org.apache.spark.util.collection.ExternalAppendOnlyMap.insertAll(ExternalAppendOnlyMap.scala:154)
	at org.apache.spark.Aggregator.combineCombinersByKey(Aggregator.scala:50)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:82)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88)
	at org.apache.spark.scheduler.Task.run(Task.scala:114)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)

17/03/16 12:46:05 WARN TaskSetManager: Lost task 0.0 in stage 5.3 (TID 71, 172.21.15.173, executor 3): FetchFailed(BlockManagerId(3, 172.21.15.173, 41404, None), shuffleId=1, mapId=8, reduceId=0, message=
org.apache.spark.shuffle.FetchFailedException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-1ce155ce-30c9-4b69-b1ea-d58e128444f1/23/shuffle_1_8_0.data, offset=0, length=40999142}
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.throwFetchFailedException(ShuffleBlockFetcherIterator.scala:416)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:356)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:58)
	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.util.CompletionIterator.hasNext(CompletionIterator.scala:32)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:39)
	at org.apache.spark.util.collection.ExternalAppendOnlyMap.insertAll(ExternalAppendOnlyMap.scala:154)
	at org.apache.spark.Aggregator.combineCombinersByKey(Aggregator.scala:50)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:82)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88)
	at org.apache.spark.scheduler.Task.run(Task.scala:114)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.IOException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-1ce155ce-30c9-4b69-b1ea-d58e128444f1/23/shuffle_1_8_0.data, offset=0, length=40999142}
	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:113)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:349)
	... 21 more
Caused by: java.io.FileNotFoundException: /tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-1ce155ce-30c9-4b69-b1ea-d58e128444f1/23/shuffle_1_8_0.data (没有那个文件或目录)
	at java.io.FileInputStream.open(Native Method)
	at java.io.FileInputStream.<init>(FileInputStream.java:146)
	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:98)
	... 22 more

)
17/03/16 12:46:05 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=31, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=65536
17/03/16 12:46:05 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=70094 lastFlushOffset=65800 createNewBlock=false
17/03/16 12:46:05 DEBUG DFSClient: Queued packet 31
17/03/16 12:46:05 DEBUG DFSClient: Waiting for ack for: 31
17/03/16 12:46:05 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 31 offsetInBlock: 65536 lastPacketInBlock: false lastByteOffsetInBlock: 70094
17/03/16 12:46:05 DEBUG DFSClient: DFSClient seqno: 31 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/16 12:46:05 WARN TaskSetManager: Lost task 1.1 in stage 5.3 (TID 73, 172.21.15.173, executor 3): FetchFailed(BlockManagerId(3, 172.21.15.173, 41404, None), shuffleId=1, mapId=3, reduceId=1, message=
org.apache.spark.shuffle.FetchFailedException: /tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-1ce155ce-30c9-4b69-b1ea-d58e128444f1/0c/shuffle_1_3_0.index (没有那个文件或目录)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.throwFetchFailedException(ShuffleBlockFetcherIterator.scala:416)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:392)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:58)
	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434)
	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440)
	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408)
	at org.apache.spark.util.CompletionIterator.hasNext(CompletionIterator.scala:32)
	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:39)
	at org.apache.spark.util.collection.ExternalAppendOnlyMap.insertAll(ExternalAppendOnlyMap.scala:154)
	at org.apache.spark.Aggregator.combineCombinersByKey(Aggregator.scala:50)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:82)
	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88)
	at org.apache.spark.scheduler.Task.run(Task.scala:114)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.FileNotFoundException: /tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-1ce155ce-30c9-4b69-b1ea-d58e128444f1/0c/shuffle_1_3_0.index (没有那个文件或目录)
	at java.io.FileInputStream.open(Native Method)
	at java.io.FileInputStream.<init>(FileInputStream.java:146)
	at org.apache.spark.shuffle.IndexShuffleBlockResolver.getBlockData(IndexShuffleBlockResolver.scala:199)
	at org.apache.spark.storage.BlockManager.getBlockData(BlockManager.scala:302)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.fetchLocalBlocks(ShuffleBlockFetcherIterator.scala:269)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.initialize(ShuffleBlockFetcherIterator.scala:303)
	at org.apache.spark.storage.ShuffleBlockFetcherIterator.<init>(ShuffleBlockFetcherIterator.scala:132)
	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:45)
	... 12 more

)
Exception in thread "main" org.apache.spark.SparkException: Job aborted due to stage failure: ResultStage 5 (treeAggregate at RowMatrix.scala:122) has failed the maximum allowable number of times: 4. Most recent failure reason: org.apache.spark.shuffle.FetchFailedException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-1ce155ce-30c9-4b69-b1ea-d58e128444f1/23/shuffle_1_8_0.data, offset=0, length=40999142} 	at org.apache.spark.storage.ShuffleBlockFetcherIterator.throwFetchFailedException(ShuffleBlockFetcherIterator.scala:416) 	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:356) 	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:58) 	at scala.collection.Iterator$$anon$12.nextCur(Iterator.scala:434) 	at scala.collection.Iterator$$anon$12.hasNext(Iterator.scala:440) 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:408) 	at org.apache.spark.util.CompletionIterator.hasNext(CompletionIterator.scala:32) 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:39) 	at org.apache.spark.util.collection.ExternalAppendOnlyMap.insertAll(ExternalAppendOnlyMap.scala:154) 	at org.apache.spark.Aggregator.combineCombinersByKey(Aggregator.scala:50) 	at org.apache.spark.shuffle.BlockStoreShuffleReader.read(BlockStoreShuffleReader.scala:82) 	at org.apache.spark.rdd.ShuffledRDD.compute(ShuffledRDD.scala:105) 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323) 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287) 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:38) 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:323) 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:287) 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:88) 	at org.apache.spark.scheduler.Task.run(Task.scala:114) 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:322) 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) 	at java.lang.Thread.run(Thread.java:745) Caused by: java.io.IOException: Error in opening FileSegmentManagedBuffer{file=/tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-1ce155ce-30c9-4b69-b1ea-d58e128444f1/23/shuffle_1_8_0.data, offset=0, length=40999142} 	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:113) 	at org.apache.spark.storage.ShuffleBlockFetcherIterator.next(ShuffleBlockFetcherIterator.scala:349) 	... 21 more Caused by: java.io.FileNotFoundException: /tmp/spark-a493e14a-5d02-4c8f-b8a9-1191b712f67f/executor-b356ffd4-0607-474d-b53b-498c27da6d6b/blockmgr-1ce155ce-30c9-4b69-b1ea-d58e128444f1/23/shuffle_1_8_0.data (没有那个文件或目录) 	at java.io.FileInputStream.open(Native Method) 	at java.io.FileInputStream.<init>(FileInputStream.java:146) 	at org.apache.spark.network.buffer.FileSegmentManagedBuffer.createInputStream(FileSegmentManagedBuffer.java:98) 	... 22 more 
	at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1456)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1444)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1443)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1443)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskCompletion(DAGScheduler.scala:1273)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:1668)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1626)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:1615)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:628)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2016)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2113)
	at org.apache.spark.rdd.RDD$$anonfun$reduce$1.apply(RDD.scala:1027)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:364)
	at org.apache.spark.rdd.RDD.reduce(RDD.scala:1009)
	at org.apache.spark.rdd.RDD$$anonfun$treeAggregate$1.apply(RDD.scala:1152)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:364)
	at org.apache.spark.rdd.RDD.treeAggregate(RDD.scala:1129)
	at org.apache.spark.mllib.linalg.distributed.RowMatrix.computeGramianMatrix(RowMatrix.scala:122)
	at org.apache.spark.mllib.linalg.distributed.RowMatrix.computeCovariance(RowMatrix.scala:344)
	at org.apache.spark.mllib.linalg.distributed.RowMatrix.computePrincipalComponentsAndExplainedVariance(RowMatrix.scala:387)
	at org.apache.spark.mllib.feature.PCA.fit(PCA.scala:48)
	at PCA.src.main.scala.PCAApp$.main(PCAApp.scala:58)
	at PCA.src.main.scala.PCAApp.main(PCAApp.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:728)
	at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:177)
	at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:202)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:116)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
17/03/16 12:46:05 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=32, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=69632
17/03/16 12:46:05 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=70094 lastFlushOffset=70094 createNewBlock=false
17/03/16 12:46:05 DEBUG DFSClient: Waiting for ack for: 31
17/03/16 12:46:05 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=33, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=69632
17/03/16 12:46:05 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=70094 lastFlushOffset=70094 createNewBlock=false
17/03/16 12:46:05 DEBUG DFSClient: Waiting for ack for: 31
17/03/16 12:46:05 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=34, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=69632
17/03/16 12:46:05 DEBUG DFSClient: DFSClient flush(): bytesCurBlock=76449 lastFlushOffset=70094 createNewBlock=false
17/03/16 12:46:05 DEBUG DFSClient: Queued packet 34
17/03/16 12:46:05 DEBUG DFSClient: Waiting for ack for: 34
17/03/16 12:46:05 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 34 offsetInBlock: 69632 lastPacketInBlock: false lastByteOffsetInBlock: 76449
17/03/16 12:46:05 DEBUG DFSClient: DFSClient seqno: 34 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/16 12:46:05 DEBUG DFSClient: DFSClient writeChunk allocating new packet seqno=35, src=/eventLogs/app-20170316124303-0007.lz4.inprogress, packetSize=65016, chunksPerPacket=126, bytesCurBlock=76288
17/03/16 12:46:05 DEBUG DFSClient: Queued packet 35
17/03/16 12:46:05 DEBUG DFSClient: Queued packet 36
17/03/16 12:46:05 DEBUG DFSClient: Waiting for ack for: 36
17/03/16 12:46:05 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 35 offsetInBlock: 76288 lastPacketInBlock: false lastByteOffsetInBlock: 76965
17/03/16 12:46:05 DEBUG DFSClient: DFSClient seqno: 35 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/16 12:46:05 DEBUG DFSClient: DataStreamer block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619 sending packet packet seqno: 36 offsetInBlock: 76965 lastPacketInBlock: true lastByteOffsetInBlock: 76965
17/03/16 12:46:05 DEBUG DFSClient: DFSClient seqno: 36 reply: SUCCESS downstreamAckTimeNanos: 0 flag: 0
17/03/16 12:46:05 DEBUG DFSClient: Closing old block BP-519507147-172.21.15.90-1479901973323:blk_1073809442_68619
17/03/16 12:46:05 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #14
17/03/16 12:46:05 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #14
17/03/16 12:46:05 DEBUG ProtobufRpcEngine: Call: complete took 18ms
17/03/16 12:46:05 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #15
17/03/16 12:46:05 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #15
17/03/16 12:46:05 DEBUG ProtobufRpcEngine: Call: getFileInfo took 1ms
17/03/16 12:46:05 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #16
17/03/16 12:46:05 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #16
17/03/16 12:46:05 DEBUG ProtobufRpcEngine: Call: rename took 4ms
17/03/16 12:46:05 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop sending #17
17/03/16 12:46:05 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop got value #17
17/03/16 12:46:05 DEBUG ProtobufRpcEngine: Call: setTimes took 6ms
17/03/16 12:46:05 DEBUG PoolThreadCache: Freed 10 thread-local buffer(s) from thread: shuffle-server-6-2
17/03/16 12:46:05 DEBUG PoolThreadCache: Freed 5 thread-local buffer(s) from thread: shuffle-server-6-3
17/03/16 12:46:05 DEBUG PoolThreadCache: Freed 5 thread-local buffer(s) from thread: shuffle-server-6-4
17/03/16 12:46:05 DEBUG PoolThreadCache: Freed 21 thread-local buffer(s) from thread: rpc-server-3-2
17/03/16 12:46:05 DEBUG PoolThreadCache: Freed 75 thread-local buffer(s) from thread: rpc-server-3-3
17/03/16 12:46:05 DEBUG Client: stopping client from cache: org.apache.hadoop.ipc.Client@7739fd54
17/03/16 12:46:05 DEBUG Client: removing client from cache: org.apache.hadoop.ipc.Client@7739fd54
17/03/16 12:46:05 DEBUG Client: stopping actual client because no more references remain: org.apache.hadoop.ipc.Client@7739fd54
17/03/16 12:46:05 DEBUG Client: Stopping client
17/03/16 12:46:05 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: closed
17/03/16 12:46:05 DEBUG Client: IPC Client (637532222) connection to spark1/172.21.15.90:9000 from hadoop: stopped, remaining connections 0
